nphysics_testbed2d/src/testbed.rs

```diff
@@ -436,25 +441,35 @@ impl State for Testbed {
         if self.running != RunMode::Stop {
             for _ in 0..self.nsteps {
                 for f in &self.callbacks {
-                    f(&mut self.world, &mut self.graphics, self.time)
+                    f(&mut self.graphics, self.time)
                 }
-                self.world.step();
+                self.world.lock().unwrap().step();
                 if !self.hide_counters {
-                    println!("{}", self.world.performance_counters());
+                    println!("{}", self.world.lock().unwrap().performance_counters());
                 }
-                self.time += self.world.timestep();
+                self.time += self.world.lock().unwrap().timestep();
             }
-
-            self.graphics.draw(&self.world, window);
         }
 
-        if self.draw_colls {
-            draw_collisions(
-                window,
-                &mut self.world,
-                &mut self.persistant_contacts,
-                self.running != RunMode::Stop,
-            );
+        // Draw scope
+        {
+            let mut physics_world = &mut self.world.lock().unwrap();
+
+            for co in physics_world.colliders() {
+                if self.graphics.body_nodes_mut(physics_world, co.data().body()).is_none() {
+                    self.graphics
+                        .add(window, co.handle(), &physics_world);
+                }
+            }
+            self.graphics.draw(&physics_world, window);
```

Comment:
This was inside of the running check because the `self.graphics.draw(...)` is actually not very well named. All it does is to synchronize the positions of the graphics objects with the position of the physics colliders. Performing this update is not necessary if the simulation is not running.